dspace-rest/src/main/java/org/dspace/rest/BitstreamResource.java

```diff
@@ -542,24 +558,9 @@ public Response updateBitstreamData(@PathParam("bitstream_id") Integer bitstream
                     headers, request, context);
 
             log.trace("Creating new bitstream.");
-            int newBitstreamId = BitstreamStorageServiceImpl.store(context, is);
-
-            log.trace("Looking for table rows of bitstreams.");
-            TableRow originalBitstreamRow = DatabaseManager.find(context, "Bitstream", bitstreamId);
-            TableRow bitstream = DatabaseManager.find(context, "Bitstream", newBitstreamId);
 
-            log.trace("Changing new internal id with old internal id.");
-            String internal_id = originalBitstreamRow.getStringColumn("internal_id");
-            Long size_bytes = originalBitstreamRow.getLongColumn("size_bytes");
-            originalBitstreamRow.setColumn("internal_id", bitstream.getStringColumn("internal_id"));
-            originalBitstreamRow.setColumn("size_bytes", bitstream.getLongColumn("size_bytes"));
-            bitstream.setColumn("internal_id", internal_id);
-            bitstream.setColumn("size_bytes", size_bytes);
-
-            DatabaseManager.update(context, originalBitstreamRow);
-            BitstreamStorageServiceImpl.delete(context, newBitstreamId);
-
-            context.complete();
+            UUID newBitstreamId = bitstreamStorageService.store(context, dspaceBitstream, is);
```

Comment:
This is a good example of the difference between a UI doing logic the wrong, complicated way, and how services can simplify this.